HermesProxy/World/Client/PacketHandlers/ItemHandler.cs

```diff
@@ -214,5 +214,36 @@ void HandleItemEnchantTimeUpdate(WorldPacket packet)
             enchant.OwnerGuid = packet.ReadGuid().To128(GetSession().GameState);
             SendPacketToClient(enchant);
         }
+
+        [PacketHandler(Opcode.SMSG_ENCHANTMENT_LOG)]
+        void HandleEnchantmentLog(WorldPacket packet)
+        {
+            EnchantmentLog enchantment = new EnchantmentLog();
+            if (LegacyVersion.AddedInVersion(ClientVersionBuild.V2_0_1_6180))
+            {
+                enchantment.Caster = packet.ReadPackedGuid().To128(GetSession().GameState);
+                enchantment.Owner = packet.ReadPackedGuid().To128(GetSession().GameState);
+            }
+            else
+            {
+                enchantment.Owner = packet.ReadPackedGuid().To128(GetSession().GameState);
+                enchantment.Caster = packet.ReadPackedGuid().To128(GetSession().GameState);
+            }
+            enchantment.ItemID = (int)packet.ReadUInt32();
+            enchantment.ItemGUID = new WowGuid128();
```

Comment:
There is no point to create a new guid here. Just delete this line. After searching for the guid below, check enchantment.ItemGUID for null, and simply return without sending the packet if the guid was not found.